akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpAttributes.scala

```diff
@@ -29,6 +30,11 @@ private[akka] object HttpAttributes {
   private[akka] def remoteAddress(address: InetSocketAddress): Attributes =
     Attributes(RemoteAddress(address))
 
+  private[akka] final case class TLSSessionInfo(session: SSLSession) extends Attribute
```

Comment:
needs comments and annotations `@InternalApi` (for tools, in the future) and `/** INTERNAL API */` (for java, since private[] is *public* in Java)

Reply:
@ktoso Thanks! Done.